File: stress/t/ctl_innerbreak.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t n=0,d=0,primes=0,firstdiv=0;
  int isprime=0,found=0;
  for(n=2;n<=60;n=n+1){
    isprime=1;
    found=0;
    d=2;
    while(d<n){
      if(found){ d=n; }
      else {
        if(n%d==0){ isprime=0; found=1; firstdiv=firstdiv+d; }
        else { d=d+1; }
      }
    }
    if(isprime){ primes=primes+1; } else { primes=primes+0; }
  }
  printf("primes="); printf("%ld",(long)primes); printf("\n");
  printf("firstdiv="); printf("%ld",(long)firstdiv); printf("\n");
  return 0;
}

File: stress/t/ctl_innerbreak.f
      PROGRAM CTLIB
      INTEGER*8 N,D,PRIMES,FDIV
      LOGICAL ISP,FOUND
      PRIMES=0
      FDIV=0
      N=2
   10 IF (N.GT.60) GOTO 40
      ISP=.TRUE.
      FOUND=.FALSE.
      D=2
   20 IF (D.GE.N) GOTO 30
      IF (FOUND) THEN
        D=N
      ELSE
        IF (MOD(N,D).EQ.0) THEN
          ISP=.FALSE.
          FOUND=.TRUE.
          FDIV=FDIV+D
        ELSE
          D=D+1
        END IF
      END IF
      GOTO 20
   30 CONTINUE
      IF (ISP) PRIMES=PRIMES+1
      N=N+1
      GOTO 10
   40 CONTINUE
      WRITE(*,'(A,I0)') 'primes=',PRIMES
      WRITE(*,'(A,I0)') 'firstdiv=',FDIV
      END
